### DISCUSSION TITLE
Putting function is MatAnisoDiffusion

### DISCUSSION THREAD

**Discussion by haqfariha** [idaholab/moose#24551] [Q&A General]
I am working on multi-phase field model.  I need to use 2 diffusion rate tensors for 2 different conditions. Am I providing the command correctly or do I need to change anything?

```
[diff_cCu]
    type = MatAnisoDiffusion
    variable = c
    diffusivity = 'if(hCu>0.8, D_bulk, D_int)' 
    v = cCu
    args = 'etaCu'
    material_property_names = 'hCu'
  [ ] 
```
While running this, I get error message- `Material property 'if(hCu>1, D_bulk, Dh2Cu)', requested by 'diff_cCu' is not defined on block Cu`
How to fix that or is there certain format to follow for **MatAnisoDiffusion**?
Thanks in advance.

**Comment by GiudGiud** [comment]
Hello

You cant put an expression for the diffusivity parameter. it has to be the name of a material property.
Please use this object instead to define that property:
https://mooseframework.inl.gov/source/materials/ParsedMaterial.html

Guillaume

**Reply by lindsayad** [reply]
[DerivativeParsedMaterial](https://mooseframework.inl.gov/source/materials/DerivativeParsedMaterial.html) may be what you're going for

**Comment by haqfariha** [comment]
I need to put condition on tensor, if it is in bulk phase, bulk diffusion rate. If not, interfacial diffusion rate. ParsedMaterial works on functions or variables maybe not on tensors. How can I do that?

**Reply by GiudGiud** [reply]
Easiest (but not so easy) way would be to make a ParsedTensorMaterial

**Reply by GiudGiud** [reply]
@dschwen is there a phase field one
